roles/ldap/tasks/main.yml

```diff
@@ -9,6 +9,24 @@
     - unzip
     - ldap-utils
 
+- name: Copying alternative service script
+  become_user: "{{ ldap_user }}"
+  become: true
+  template:
+    src: "templates/{{ item.src }}"
+    dest: "/etc/init.d/"
+    mode: 0755
+  with_items:
+    - { src: "slapd.j2", dest: "slapd" }
+  when: ansible_virtualization_type == "docker"
+
+- name: Reload systemctl
+  command: systemctl daemon-reload
+  when: ansible_virtualization_type == "docker"
+
+- name: Starting and enabling the LDAP service
+  systemd: daemon_reload=yes name=slapd state=restarted enabled=yes
```

Comment:
Please split this in two parts: a rule that enables and starts slapd (and correctly does nothing if slapd is already runnign and enabled), and a handler/notifier that restarts slaps if the configuration changes